### PR TITLE
Make APIv6 client retry strategy configurable

### PR DESCRIPTION
**Pull request by Fenthick**
Issue: KNTK-261

### CODE REVIEW

**Review by mmac-m3a — APPROVED** [review]

File: apiv6/kentikapi/httputil/httputil.go [hunk 1]
```diff
@@ -39,8 +37,141 @@ func NewRetryingClient(cfg ClientConfig) *http.Client {
...
+	c.CheckRetry = makeRetryPolicy(cfg.RetryableStatusCodes, cfg.RetryableMethods)
 	c.ErrorHandler = retryablehttp.PassthroughErrorHandler
 
 	return c.StandardClient()
 }
+
+// ClientConfig holds configuration for retrying client.
+type ClientConfig struct {
+	HTTPClient           *http.Client
+	RetryMax             *int
```
Comment (mateuszmidor) [inline apiv6/kentikapi/httputil/httputil.go:49]: The Config and ClientConfig retry-related attributes could be extracted into dedicated struct like HTTPRetryConfig as they repeat and make pretty cohesive, single-purpose pack that could be documented in single place

File: apiv6/kentikapi/httputil/httputil.go [hunk 2]
```diff
@@ -39,8 +37,141 @@ func NewRetryingClient(cfg ClientConfig) *http.Client {
...
+		return false, nil
+	}
+}
+
+func isRequestRetrayable(r *http.Request, methodsSet map[string]struct{}) bool {
+	_, ok := methodsSet[r.Method]
+	return ok
+}
+
+func isErrorRecoverable(err error, redirectsErrorRe, schemeErrorRe *regexp.Regexp) (bool, error) {
```
Comment (mateuszmidor) [inline apiv6/kentikapi/httputil/httputil.go:127]: Telling errors one from another by parsing error message can prove brittle as the message can change without notice, is there maybe some more robust way?

**Review by mateuszmidor — COMMENTED** [review]
Nice job, only the error message parsing worries me a bit, but except for that - thumb up.

File: apiv6/kentikapi/httputil/httputil.go (hunk 2 above)
Comment (Fenthick) [reply apiv6/kentikapi/httputil/httputil.go:127]: Fixed in https://github.com/kentik/community_sdk_golang/pull/42

**Pull request merged**